batched/dense/impl/KokkosBatched_Trsm_Serial_Internal.hpp

```diff
@@ -34,47 +36,74 @@ namespace KokkosBatched {
 template <typename AlgoType>
 struct SerialTrsmInternalLeftLower {
   template <typename ScalarType, typename ValueType>
-  KOKKOS_INLINE_FUNCTION static int invoke(const bool use_unit_diag, const int m, const int n, const ScalarType alpha,
-                                           const ValueType *KOKKOS_RESTRICT A, const int as0, const int as1,
+  KOKKOS_INLINE_FUNCTION static int invoke(const bool use_unit_diag, const bool do_conj, const int m, const int n,
+                                           const ScalarType alpha, const ValueType *KOKKOS_RESTRICT A, const int as0,
+                                           const int as1,
                                            /**/ ValueType *KOKKOS_RESTRICT B, const int bs0, const int bs1);
 };
 
 template <>
 template <typename ScalarType, typename ValueType>
 KOKKOS_INLINE_FUNCTION int SerialTrsmInternalLeftLower<Algo::Trsm::Unblocked>::invoke(
-    const bool use_unit_diag, const int m, const int n, const ScalarType alpha, const ValueType *KOKKOS_RESTRICT A,
-    const int as0, const int as1,
+    const bool use_unit_diag, const bool do_conj, const int m, const int n, const ScalarType alpha,
+    const ValueType *KOKKOS_RESTRICT A, const int as0, const int as1,
     /**/ ValueType *KOKKOS_RESTRICT B, const int bs0, const int bs1) {
   const ScalarType one(1.0), zero(0.0);
 
   if (alpha == zero)
     KokkosBlas::Impl::SerialSetInternal::invoke(m, n, zero, B, bs0, bs1);
   else {
     if (alpha != one) KokkosBlas::Impl::SerialScaleInternal::invoke(m, n, alpha, B, bs0, bs1);
-    if (m <= 0 || n <= 0) return 0;
 
-    for (int p = 0; p < m; ++p) {
-      const int iend = m - p - 1, jend = n;
+    if (do_conj) {
+      for (int p = 0; p < m; ++p) {
+        const int iend = m - p - 1, jend = n;
 
-      const ValueType *KOKKOS_RESTRICT a21 = iend ? A + (p + 1) * as0 + p * as1 : NULL;
+        const ValueType *KOKKOS_RESTRICT a21 = iend ? A + (p + 1) * as0 + p * as1 : nullptr;
 
-      ValueType *KOKKOS_RESTRICT b1t = B + p * bs0, *KOKKOS_RESTRICT B2 = iend ? B + (p + 1) * bs0 : NULL;
+        ValueType *KOKKOS_RESTRICT b1t = B + p * bs0, *KOKKOS_RESTRICT B2 = iend ? B + (p + 1) * bs0 : nullptr;
 
-      if (!use_unit_diag) {
-        const ValueType alpha11 = A[p * as0 + p * as1];
+        if (!use_unit_diag) {
+          const ValueType alpha11 = Kokkos::ArithTraits<ValueType>::conj(A[p * as0 + p * as1]);
 
 #if defined(KOKKOS_ENABLE_PRAGMA_UNROLL)
 #pragma unroll
 #endif
-        for (int j = 0; j < jend; ++j) b1t[j * bs1] = b1t[j * bs1] / alpha11;
+          for (int j = 0; j < jend; ++j) b1t[j * bs1] = b1t[j * bs1] / alpha11;
+        }
+
+        for (int i = 0; i < iend; ++i)
+
+#if defined(KOKKOS_ENABLE_PRAGMA_UNROLL)
+#pragma unroll
+#endif
+          for (int j = 0; j < jend; ++j)
+            B2[i * bs0 + j * bs1] -= Kokkos::ArithTraits<ValueType>::conj(a21[i * as0]) * b1t[j * bs1];
```

Comment:
```suggestion
            B2[i * bs0 + j * bs1] -= (do_conj ? Kokkos::ArithTraits<ValueType>::conj(a21[i * as0]) * b1t[j * bs1] : a21[i * as0] * b1t[j * bs1]);
```

With this you can remove a lot of the code duplication introduce for the conjugate type.